src/components/Event/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { castDraft } from 'immer';
+import React, { useCallback, useContext, useState } from 'react';
+import {
+  faPencil,
+  faPalette,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons';
+
+import { classes, getContentClassName, periodToString } from '../../utils/misc';
+import { ActionRow, Palette } from '..';
+import { ScheduleContext } from '../../contexts';
+import { Period } from '../../types';
+
+import './stylesheet.scss';
+
+export type CourseProps = {
+  eventId: string;
+  eventName: string;
+  eventPeriod: Period;
+  eventDays: string[];
```

Comment:
I think we should make this readonly, since that is the type passed in my ScheduleContext